From f697594c1a3f8b5c5745382628b9b781d605ac88 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 16 Jan 2021 14:58:59 -0500
Subject: [PATCH 20/20] sepolicy/init: add rule to fix cam2api option in
 trebleapp

  logs:
      17 00:47:48.796     1     1 W /system/bin/init: type=1107
      audit(0.0:357): uid=0 auid=4294967295 ses=4294967295
      subj=u:r:init:s0 msg='avc: denied { set } for
      property=persist.vendor.camera.eis.enable pid=4411 uid=1000
      gid=1000 scontext=u:r:system_app:s0
      tcontext=u:object_r:vendor_default_prop:s0 tclass=property_service
      permissive=0'

      01-17 00:47:48.796     1     1 W /system/bin/init: type=1107
      audit(0.0:356): uid=0 auid=4294967295 ses=4294967295
      subj=u:r:init:s0 msg='avc: denied { set } for
      property=persist.vendor.camera.HAL3.enabled pid=4411 uid=1000
      gid=1000 scontext=u:r:system_app:s0
      tcontext=u:object_r:vendor_default_prop:s0 tclass=property_service
      permissive=0'
---
 sepolicy/init.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sepolicy/init.te b/sepolicy/init.te
index 7fb7079..8bfb8fc 100644
--- a/sepolicy/init.te
+++ b/sepolicy/init.te
@@ -8,3 +8,5 @@ allow init userdata_block_device:lnk_file relabelto;
 allow vendor_init vendor_init:capability { sys_module };
 
 allow init system_file:lnk_file create_file_perms;
+
+allow init vendor_default_prop:property_service { set };
-- 
2.17.1

